src/DI/EventDispatcherExtension.php

```diff
@@ -45,6 +50,33 @@ public function loadConfiguration(): void
 			$eventDispatcherDefinition
 				->setFactory(EventDispatcher::class);
 		}
+
+		if (!$this->config->debug && $this->config->logger === null) {
+			return;
+		}
+
+		$wrapped = $builder->addDefinition($this->prefix('wrappedOriginal'))
```

Comment:
I'd prefer sth like that:

```
if (debug) {
  // register DiagnosticDispatcher ( normal Dispatcher )
} else {
  // register normal EventDispatcher
}
```

Reply:
OK, i've realized it's not possible to use own listener anyway so it's always one of classes provided by this extension.

Reply:
I don't know how to write the factory function for diagnostic version without creating second service for the wrapped dispatcher, as the LazyDispatcher requires container in constructor.
so is something like this ok?

```suggestion
		if ($config->lazy === true) {
			$factory = LazyEventDispatcher::class;
		} else {
			$factory = EventDispatcher::class;
		}

		if ($this->config->debug || $this->config->logger !== null) {
			$inner = $builder->addDefinition($this->prefix('innerDispatcher'))
				->setAutowired(false)
				->setFactory($factory);
			$eventDispatcherDefinition->setFactory(DiagnosticDispatcher::class, [$inner]);
		} else {
			$eventDispatcherDefinition->setFactory($factory);
		}
```